src/input/devices.rs

```diff
@@ -120,16 +173,98 @@ impl TrackedDevice {
         }
     }
 
-    pub fn get_type(&self) -> TrackedDeviceType {
-        self.device_type
+    pub fn get_type(&self) -> &TrackedDeviceType {
+        &self.device_type
     }
 
     pub fn get_controller_hand(&self) -> Option<Hand> {
-        match self.get_type() {
+        match self.device_type {
             TrackedDeviceType::Controller { hand, .. } => Some(hand),
             _ => None,
         }
     }
+
+    fn get_string_property(&self, property: vr::ETrackedDeviceProperty) -> Option<&CStr> {
+        let hand = match self.device_type {
+            TrackedDeviceType::Controller { hand } => hand,
+            _ => Hand::Left,
+        };
+
+        let data = self.interaction_profile?.properties();
+
+        match property {
+            // Audica likes to apply controller specific tweaks via this property
+            vr::ETrackedDeviceProperty::ControllerType_String => Some(data.openvr_controller_type),
+            // I Expect You To Die 3 identifies controllers with this property -
+            // why it couldn't just use ControllerType instead is beyond me...
+            // Because some controllers have different model names for each hand......
+            vr::ETrackedDeviceProperty::ModelNumber_String => Some(*data.model.get(hand)),
+            // Resonite won't recognize controllers without this
+            vr::ETrackedDeviceProperty::RenderModelName_String => {
+                Some(*data.render_model_name.get(hand))
+            }
+            vr::ETrackedDeviceProperty::RegisteredDeviceType_String => {
+                Some(*data.registered_device_type.get(hand))
+            }
+            vr::ETrackedDeviceProperty::TrackingSystemName_String => {
+                Some(data.tracking_system_name)
+            }
+            // Required for controllers to be acknowledged in I Expect You To Die 3
+            vr::ETrackedDeviceProperty::SerialNumber_String => match self.get_type() {
+                TrackedDeviceType::Controller { .. } => Some(*data.serial_number.get(hand)),
+                #[cfg(feature = "monado")]
+                TrackedDeviceType::GenericTracker { serial, .. } => Some(serial.as_c_str()),
+                _ => None,
```

Comment:
```suggestion
                TrackedDeviceType::Hmd => unreachable!()
```
Force handling if we add any other device types in the future.